controllers/etcd_controller.go

```diff
@@ -939,6 +729,14 @@ func (r *EtcdReconciler) getMapFromEtcd(im imagevector.ImageVector, etcd *druidv
 		// "password":                etcd.Spec.Etcd.Password,
 	}
 
+	if etcd.Spec.Etcd.ClientPort != nil {
+		etcd.Spec.Etcd.ClientPort = pointer.Int32Ptr(val.Service.ClientPort)
+	}
+
+	if etcd.Spec.Etcd.ServerPort != nil {
+		etcd.Spec.Etcd.ServerPort = pointer.Int32Ptr(val.Service.ServerPort)
+	}
+
```

Comment:
Why is the `etcd` object changed here?

Reply:
It was needed because statefulset needs the value of ports from ETCD spec. I needed to pass the port values to syncStatefulset() function. For that, either I needed to pass the port values as arguments to GenerateValues() function or pass it through ETCD spec. I took the later approach as the earlier one would need to write a function with long parameter list.

Reply:
I'd nevertheless propose to do it via the GenerateValues() function because changing the `etcd` object in place can cause unexpected results. So if you want to actually write back the port that we should also consider updating the etcd object at the end. If this is not intended or necessary then we should also not manipulate the object.

Reply:
Okay. Addressed